service/history/replication/raw_task_converter.go

```diff
@@ -253,6 +256,9 @@ func convertSyncHSMReplicationTask(
 
 			stateMachineNode := common.CloneProto(mutableState.HSM().InternalRepr())
 			workflow.SanitizeStateMachineNode(stateMachineNode)
+			if err := common.DiscardUnknownProto(stateMachineNode); err != nil {
```

Comment:
do you think it is worth just do this in CloneProto?

Reply:
hmm I'd like to keep those utilities separate. Carrying over unknown fields is the default behavior of proto, so I don't want to introduce any surprise for anyone using the clone method.